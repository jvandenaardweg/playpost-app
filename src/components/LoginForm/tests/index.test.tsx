import React from 'react';
import { fireEvent, render, RenderAPI } from 'react-native-testing-library';

import { LoginForm, Props } from '../index';

const onPressLoginHandler = jest.fn();
const onChangeTextHandler = jest.fn();
const onPressForgotPasswordHandler = jest.fn();

const defaultProps: Props = {
  onChangeText: onChangeTextHandler,
  onPressLogin: onPressLoginHandler,
  onPressForgotPassword: onPressForgotPasswordHandler,
  email: '',
  password: '',
  isLoading: false
}
describe('LoginForm', () => {
  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeEach(() => {
      const props = {
        ...defaultProps
      }
      wrapper = render(<LoginForm {...props} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('interacting', () => {
    let wrapper: RenderAPI;

    beforeEach(() => {
      const props = {
        ...defaultProps
      }
      wrapper = render(<LoginForm {...props} />);
    });

    it('should fire onPressLogin when the login button is pressed', () => {
      fireEvent.press(wrapper.getByTestId('LoginForm-Button-login'));
      expect(onPressLoginHandler).toHaveBeenCalledTimes(1);
    });

    it('should fire onPressForgotPassword when the forgot password button is pressed', () => {
      fireEvent.press(wrapper.getByTestId('LoginForm-Button-forgot-password'));
      expect(onPressForgotPasswordHandler).toHaveBeenCalledTimes(1);
    });

    it('should fire onChangeText when filling in a TextInput', () => {
      const exampleEmail = 'jordyvandenaardweg@gmail.com';
      const emailTextInput = wrapper.getByTestId('LoginForm-TextInput-email');

      fireEvent.changeText(emailTextInput, exampleEmail);
      expect(onChangeTextHandler).toHaveBeenCalledWith('email', exampleEmail);
    });

    it('should fire onChangeTextHandler when filling in a TextInput', () => {
      const examplePassword = 'p@ssword!';
      const passwordTextInput = wrapper.getByTestId('LoginForm-TextInput-password');

      fireEvent.changeText(passwordTextInput, examplePassword);
      expect(onChangeTextHandler).toHaveBeenCalledWith('password', examplePassword);
    });

    it('should disable the signup button if isLoading is true', () => {
      const props = {
        ...defaultProps,
        isLoading: true
      }

      wrapper.update(<LoginForm {...props} />);

      expect(wrapper.getByTestId('LoginForm-Button-login').props.disabled).toBe(true);
    });
  });
});
