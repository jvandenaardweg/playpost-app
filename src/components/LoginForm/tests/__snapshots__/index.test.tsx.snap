// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm rendering should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="LoginForm"
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "100%",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "marginBottom": 14,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 15,
                "lineHeight": 23,
              },
              Object {
                "marginBottom": 4,
              },
            ]
          }
          testID="InputGroup-Text-label"
        >
          E-mail address
        </Text>
        <TextInput
          autoCapitalize="none"
          autoFocus={false}
          blurOnSubmit={false}
          clearButtonMode="always"
          editable={true}
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="john.appleseed@playpost.app"
          returnKeyType="next"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#EEEEEE",
              "borderRadius": 6,
              "fontFamily": "Inter-Regular",
              "fontSize": 15,
              "height": 50,
              "lineHeight": undefined,
              "paddingLeft": 14,
              "paddingRight": 14,
            }
          }
          testID="LoginForm-TextInput-email"
          textContentType="username"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 14,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#EEEEEE",
              "bottom": 0,
              "height": 50,
              "justifyContent": "center",
              "position": "absolute",
              "right": 4,
              "width": 50,
              "zIndex": 1,
            }
          }
          testID="InputGroup-View-right-element-container"
        >
          <View
            accessible={true}
            clickable={true}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="InputGroupPassword-TouchableOpacity-toggle-password-visible"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 20,
                  },
                  Object {
                    "height": 20,
                    "width": 20,
                  },
                  Object {
                    "fontFamily": "Feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="InputGroupPassword-TouchableOpacity-toggle-password-visible-icon"
            >
              ÔÖù
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 15,
                "lineHeight": 23,
              },
              Object {
                "marginBottom": 4,
              },
            ]
          }
          testID="InputGroup-Text-label"
        >
          Password
        </Text>
        <TextInput
          autoCapitalize="none"
          autoFocus={false}
          blurOnSubmit={false}
          clearButtonMode="always"
          editable={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Your super secret password"
          returnKeyType="done"
          secureTextEntry={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#EEEEEE",
              "borderRadius": 6,
              "fontFamily": "Inter-Regular",
              "fontSize": 15,
              "height": 50,
              "lineHeight": undefined,
              "paddingLeft": 14,
              "paddingRight": 14,
            }
          }
          testID="LoginForm-TextInput-password"
          textContentType="password"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 3,
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="LoginForm-Button-login"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Login
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 3,
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="LoginForm-Button-forgot-password"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#2089dc",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Forgot password?
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
